packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -181,6 +222,7 @@ class DayPicker extends StatelessWidget {
     @required this.firstDate,
     @required this.lastDate,
     @required this.displayedMonth,
+    this.onHeaderTap,
```

Comment:
OK, I understand why this is needed. 

A slightly more specific name would be helpful. Like onMonthHeaderTap